react/components/ShippingSimulator/index.js

```diff
@@ -1,97 +1,54 @@
-import React, { Component, Fragment } from 'react'
+import React, { Fragment, useState } from 'react'
 import { injectIntl, intlShape } from 'react-intl'
 import { compose, withApollo } from 'react-apollo'
-import ContentLoader from 'react-content-loader'
 import PropTypes from 'prop-types'
-import { Button, Input } from 'vtex.styleguide'
+import { Button } from 'vtex.styleguide'
+import {
+  AddressRules,
+  AddressContainer,
+  PostalCodeGetter
+} from 'vtex.address-form'
+import { StyleguideInput } from 'vtex.address-form/inputs'
+import { addValidation, removeValidation } from 'vtex.address-form/helpers'
 
 import ShippingTable from './components/ShippingTable'
 import getShippingEstimates from './queries/getShippingEstimates.gql'
-
-import styles from './styles.css'
-/**
- * Shipping simulator component
- *
- * Display an input for the zipcode
- */
-class ShippingSimulator extends Component {
-  static propTypes = {
-    intl: intlShape.isRequired,
-    client: PropTypes.object,
-    skuId: PropTypes.string,
-    seller: PropTypes.string,
-    country: PropTypes.string.isRequired,
-    /** Component and content loader styles */
-    styles: PropTypes.object,
-  }
-
-  static Loader = (loaderProps = {}) => (
-    <div className={`${styles.shippingContainer}`}>
-      <ContentLoader
-        className={`${styles.shippingContainerLoader}`}
-        style={{
-          width: '100%',
-          height: '100%',
-        }}
-        width={500}
-        height={40}
-        preserveAspectRatio="xMinYMin meet"
-        {...loaderProps}
-      >
-        <rect
-          height="100%"
-          width="7em"
-          {...loaderProps[`${styles.shippingZipcodeLabelLoader}`]}
-        />
-        <rect
-          height="100%"
-          width="15em"
-          x="8em"
-          {...loaderProps[`${styles.shippingInputLoader}`]}
-        />
-      </ContentLoader>
-    </div>
-  )
-
-  state = {
-    zipcodeValue: '',
-    prevZipcode: '',
-    shipping: {},
-    loading: false,
-  }
-
-  handleChange = e => {
-    let zipcodeValue = e.target.value || ''
-
-    zipcodeValue = zipcodeValue.replace(/\D/g, '')
-
-    if (zipcodeValue.length > 8) {
-      zipcodeValue = zipcodeValue.substring(0, 8)
-    }
-
-    if (zipcodeValue.length > 5) {
-      zipcodeValue = zipcodeValue.replace(/(\d{5})(\d+)/, '$1-$2')
-    }
-
-    this.setState({ zipcodeValue })
+import ShippingSimulatorLoader from './Loader'
+import shippingSimulator from './shippingSimulator.css'
+import { getNewAddress } from './utils'
+
+const ShippingSimulator = ({
+  intl,
+  client,
+  skuId,
+  seller,
+  country,
+  styles,
+}) => {
+  const [address, setAddress] = useState(addValidation(getNewAddress(country)))
```

Comment:
use the lazy initialization, so you don't call `getNewAddress` on every render
```suggestion
  const [address, setAddress] = useState(() => addValidation(getNewAddress(country)))
```